Source/pack.cpp

```diff
@@ -334,8 +334,8 @@ void UnPackItem(const ItemPack &packedItem, const Player &player, Item &item, bo
 		item = {};
 		// Item generation logic will assign CF_HELLFIRE based on isHellfire
 		// so if we carry it over from packedItem, it may be incorrect
-		item.dwBuff = SDL_SwapLE32(packedItem.dwBuff) & ~CF_HELLFIRE;
-		RecreateItem(player, item, idx, SDL_SwapLE16(packedItem.iCreateInfo), SDL_SwapLE32(packedItem.iSeed), SDL_SwapLE16(packedItem.wValue), isHellfire);
+		const uint32_t dwBuff = (SDL_SwapLE32(packedItem.dwBuff) & ~CF_HELLFIRE) | (isHellfire ? CF_HELLFIRE : 0);

```

Comment:
We can simultaneously simplify this logic and correctly deduce the appropriate flag more often if we don't clear the flag from `dwBuff` and simply OR these together.
```suggestion
		const uint32_t dwBuff = SDL_SwapLE32(packedItem.dwBuff) | (isHellfire ? CF_HELLFIRE : 0);
```